### PR TITLE
Regression tests for consensus critical parts

### PR DESCRIPTION
**Pull request by aslesarenko**

### CODE REVIEW

File: sigmastate/src/test/scala/special/sigma/SigmaDslTest.scala [hunk 1]
```diff
@@ -663,17 +695,24 @@ class SigmaDslTest extends PropSpec
...
+      eq(Colls.fromArray(x))
     }
   }
 
+  val arrayWithRangeGen = for {
+    arr <- arrayGen[Int];
+    l <- Gen.choose(0, arr.length - 1);
+    r <- Gen.choose(l, arr.length - 1) } yield (arr, (l, r))
+
   property("Coll patch method equivalnce") {
```
Comment (kushti) [inline sigmastate/src/test/scala/special/sigma/SigmaDslTest.scala:802]: equivalnce => equivalence

File: sigmastate/src/test/scala/special/sigma/SigmaDslTest.scala [hunk 2]
```diff
@@ -685,22 +724,32 @@ class SigmaDslTest extends PropSpec
...
       whenever (size > 1) {
-        val index = getRandomIndex(size)
-        eq(Builder.DefaultCollBuilder.fromArray(x._1), (index, x._2))
+        val index = getArrayIndex(size)
+        eq(Colls.fromArray(x._1), (index, x._2))
       }
     }
   }
 
   property("Coll updateMany method equivalnce") {
```
Comment (kushti) [inline sigmastate/src/test/scala/special/sigma/SigmaDslTest.scala:828]: equivalnce => equivalence

File: sigmastate/src/test/scala/sigmastate/ErgoTreeSpecification.scala [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+          tyDesc.methods.length shouldBe methods.length
+          for (expectedMethod <- methods) {
+            if (expectedMethod.isResolvableFromIds) {
+
+              // the following line is used in MethodCall deserializer to resolve SMethod
+              val resolvedMethod = SMethod.fromIds(typeId, expectedMethod.methodId)
+
+              resolvedMethod.objType.typeId shouldBe typeId
+              resolvedMethod.name shouldBe expectedMethod.method.name
+//              assert(resolvedMethod.irInfo.irBuilder.isDefined, s"irBuilder is not defined for $resolvedMethod")
```
Comment (kushti) [inline sigmastate/src/test/scala/sigmastate/ErgoTreeSpecification.scala:243]: Please avoid commented out code

File: sigmastate/src/test/scala/special/sigma/SigmaDslTest.scala [hunk 4]
```diff
@@ -5,6 +5,8 @@ import java.math.BigInteger
 import org.ergoplatform.ErgoScriptPredef.TrueProp
 import org.ergoplatform.dsl.{SigmaContractSyntax, TestContractSpec}
 import org.ergoplatform._
+import org.scalacheck.Gen
+import org.scalatest.exceptions.TestFailedException
```
Comment (kushti) [inline sigmastate/src/test/scala/special/sigma/SigmaDslTest.scala:10]: Added import is not used

File: sigmastate/src/test/scala/special/sigma/SigmaDslTest.scala (hunk 1 above)
Comment (aslesarenko) [reply sigmastate/src/test/scala/special/sigma/SigmaDslTest.scala:800]: fixed

File: sigmastate/src/test/scala/special/sigma/SigmaDslTest.scala (hunk 2 above)
Comment (aslesarenko) [reply sigmastate/src/test/scala/special/sigma/SigmaDslTest.scala:826]: fixed

File: sigmastate/src/test/scala/special/sigma/SigmaDslTest.scala (hunk 4 above)
Comment (aslesarenko) [reply sigmastate/src/test/scala/special/sigma/SigmaDslTest.scala]: fixed

File: sigmastate/src/test/scala/sigmastate/ErgoTreeSpecification.scala (hunk 3 above)
Comment (aslesarenko) [reply sigmastate/src/test/scala/sigmastate/ErgoTreeSpecification.scala]: fixed

**Pull request merged**